chapters/methodology.tex

```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying the configuration file of the app. A configuration-based 
+factory will create the Face- or Smile-Detector based on a configuration file. 
+(See \ref{fig:arch1})
+
+The Image Analyzer uses the created implementations of Face- and Smile-
+Detectors. The results of the Image Analyzer are collected and stored in a
+local database.
+
+In order to load and manage the configuration file, an configuration manager 
+is used.
+
+\begin{figure}
+    \includegraphics[width=\linewidth]{figures/methodology_architecture_1.jpg}
+    \caption{class diagram describing the architecture}

```

Comment:
Capitalize "class"

Something I didn't realize at first as well is that you must briefly describe every figure in your text, so e. g.: "Figure `\ref{fig:arch1}` shows the class diagram of the architecture". You already have "(See `\ref{fig:arch1}`)" above, but you probably have to provide a more detailed description.